### ISSUE TITLE
"connected" and "related" should not reference themselves

### ISSUE THREAD

**Issue by jketreno** [Board-Explorer/board-viewer#20]
J3 Pin 5 on the D2000 is connected to J13 Pin 4. As of 0.0.25, the "connected" array for J3Pin5 includes a reference to itself; it should only include J13Pin4.

At the end of the lists being built, all the items should be passed over to remove self references.

**Comment by juliocri** [comment]
Related are still adding references to themselves, example in D200 with VBAT.

**Comment by jketreno** [comment]
Test case:

Incoming JSON:
```json
{
  "J4": {
    "5": {
      "connected": [ "J8Pin8" ]
    }
  },
  "J8": {
    "8": {}
  },
  "SW2": {
    "related": [ "J4Pin5" ]
  }
}
```

Processed associations should be:

```json
{
  "J4": {
    "5": {
      "connected": [ "J8Pin8" ],
      "related": [ "SW2" ]
    }
  },
  "J8": {
    "8": {
      "connected": [ "J4Pin5" ],
      "related": [ "SW2" ]
    }
  },
  "SW2": {
    "related": [ "J4Pin5", "J8Pin8" ]
  }
}
```

**Comment by jketreno** [comment]
Fixed and functional as of 0.0.30
